src/languages/stan.js

```diff
@@ -436,30 +436,36 @@ export default function(hljs) {
         }
       },
       {
-        // highlights distributrions that end with special endings
+        // highlights distributions that end with special endings
         scope: 'built_in',
-        match: /\w+(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/
-      },
-      {
-        // cannot ever have these endings with the ~
-        match: [
-          /~\s*/,
-          /\w+(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/
-        ],
-        scope: { 2: "illegal" }
+        keywords: DISTRIBUTIONS,
+        begin: regex.concat(/\w*/, regex.either(...DISTRIBUTIONS), /(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/)
       },
       {
-        // highlights user defined distributions
-        match: [
+        // highlights user defined distributions after ~
+        begin: [
           /~/,
           /\s*/,
           /\w+(?=\s*[\(.*\)])/
         ],
-        scope: { 3: "built_in" },
+        scope: { 3: "title.function" },
+      },
+      {
+        // highlights user defined distributions with special endings
+        scope: 'title.function',
+        begin: /\w*(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/
+      },
+      {
+        // cannot ever have these endings with the ~
+        begin: [
+          /~\s*/,
+          /\w*(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/
+        ],
+        scope: { 2: "no-highlight" }
```

Comment:
This is not  valid scope... if you don't want to highlight them, then just have no scope at all.  But what is this rule accomplishing, acting as a guard for which other rule?

Reply:
it's acting as a guard to not highlight distribution that has those endings *after* a `~`. So should I just delete line 464?

Reply:
> it's acting as a guard to not highlight distribution that has those endings after a ~

A comment would be helpful to explain this.

But what is the deal with distributions that come after `~`? (curious)

> So should I just delete line 464?

Yes, though then you may need to collapse the multi-matcher to a  single regex, can't recall if it'll force ya to do that or not.

Reply:
I removed the scope. I had to keep the `scope: {}` just without anything inside or else it breaks.

Reply:
> But what is the deal with distributions that come after `~`? (curious)

it's basically syntactic sugar for an unnormalized log density. Stan didn't have `_lupdf` before to indicate this so if you wanted an unnormalized density (which is faster but you cannot do certain things with the posterior samples, like the bridgesampling R package needs).

Reply:
> it's basically syntactic sugar for an unnormalized log density.

Why do we not want to highlight "unnormalized log density" distributions?

Reply:
You just can't have an ending `_lpdf, lcdf, lupdf, etc.` with the ~. You can only do it with `target +=` hence it should not be highlighted.

Reply:
You can either do
```stan
target += some_density_in_stan_lupdf(y | x);
```
or
```
y ~ some_density_in_stan(x);
```
you cannot write
```
y ~ some_density_in_stan_lupdf( ? | x);
```

Reply:
> you cannot write

_So why would it exist in the code at all as something we need to match for?_ - why would it be something we have to worry about at all then?

Reply:
I just didn't want to highlight something that was invalid. Otherwise this will get highlighted as valid.

Reply:
The issue is that the distributions are keywords and these will match regardless of where they are. I need them to match in many places except you cannot have a distribution with those endings after a ~.

Reply:
I think I'm going to yank this (it can be revisited in the future with more data or concrete examples).  Our job isn't to decide if code is "valid" or not...  we highlight *correct* code - it is expected that code we are asked to highlight will be "correct".  Results for *incorrect* code are unpredictable.

We do sometimes try to avoid false positives (with guards like this), but not to the extent of going out of our way to avoid matching code we KNOW is invalid.

If someone says "hey this shouldn't be highlighted" we say "sorry, but your code is buggy, so fix it first".

Reply:
cool, sounds good